adapters/sovrn/sovrn.go

```diff
@@ -0,0 +1,274 @@
+package sovrn
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbs"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+type SovrnAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+// Name - export adapter name */
+func (s *SovrnAdapter) Name() string {
+	return "sovrn"
+}
+
+// FamilyName used for cookies and such
+func (s *SovrnAdapter) FamilyName() string {
+	return "sovrn"
+}
+
+type sovrnParams struct {
+	TagID int `json:"tagid"`
+}
+
+func (s *SovrnAdapter) SkipNoCookies() bool {
+	return false
+}
+
+// Call send bid requests to sovrn and receive responses
+func (s *SovrnAdapter) Call(ctx context.Context, req *pbs.PBSRequest, bidder *pbs.PBSBidder) (pbs.PBSBidSlice, error) {
+	supportedMediaTypes := []pbs.MediaType{pbs.MEDIA_TYPE_BANNER}
+	sReq, err := adapters.MakeOpenRTBGeneric(req, bidder, s.FamilyName(), supportedMediaTypes, true)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sovrnReq := openrtb.BidRequest{
+		ID:   sReq.ID,
+		Imp:  sReq.Imp,
+		Site: sReq.Site,
+	}
+
+	// add tag ids to impressions
+	for i, unit := range bidder.AdUnits {
+		var params sovrnParams
+		err = json.Unmarshal(unit.Params, &params)
+		if err != nil {
+			return nil, err
+		}
+		sovrnReq.Imp[i].Secure = sReq.Imp[i].Secure
+		sovrnReq.Imp[i].TagID = strconv.Itoa(params.TagID)
+		sovrnReq.Imp[i].Banner.Format = nil // todo: need to do this?
```

Comment:
Don't leave TODO's hanging around in the code, unless you also open a github issue for them and use `// TODO #XYZ: {whatever}`

In this case it sounds like you intended to figure this out before the initial merge. If you've already done that, remove the comment.